base_layer/core/src/validation/transaction_validators.rs

```diff
@@ -76,6 +76,33 @@ impl<B: BlockchainBackend> Validation<Transaction, B> for FullTxValidator<B> {
     }
 }
 
+/// This validator assumes that the transaction was already validated and it will skip this step. It will only check, in
+/// order,: All inputs exist in the backend, All timelocks (kernel lock heights and output maturities) have passed
+pub struct OrphanpoolTxValidator<B: BlockchainBackend> {
```

Comment:
I think we can get a more accurate name here. See if you can improve on my suggestion,
```suggestion
pub struct TxIsSpendableValidator<B: BlockchainBackend> {
```
Alts:
* TxIsSpendable
* TxInputAndMaturityValidator